proposals/aip-1.md

```diff
@@ -0,0 +1,128 @@
+---
+aip: 1
+title: Proposal Purpose and Guidelines
+status: Draft
+type: Process
+author: Scott Willeke (scott@willeke.com)
+created: 2018-04-01
+---
+
+# What is an Agorex Improvement Proposal (AIP)? #
+An AIP is a document providing information to the Agorex community or describing process that the Agorex community follows. The AIP should provide a concise specification of a process and a rationale for it. The AIP author is responsible for building consensus within the community and documenting dissenting opinions.
+
+
+# Rational #
+To encourage and support our goal of providing a public, open, honest, community-run decentralized exchange, we need a well defined process for collecting, discussing, and adopting processes that the Board of Directors executes on behalf of the community.
+
+We intend for AIPs to be the primary mechanisms for the community to propose new processes, raise issues, and collect community input on processes and issues. AIPs serve as the official documentation for the the processes that the Board of Directors executes on behalf of the community. Because the AIPs are maintained as text files in a versioned repository, their revision history is the historical record of the proposal.
+
+
+# Proposal Types #
+There are currently two types of proposals:
+
+- An **Informational Proposal** describes an issue, or provides general guidelines or information to the community. Informational proposals do not necessarily represent community consensus or a recommendation, so community members and users are free to ignore Informational Proposals or follow their advice.
+
+- A **Process Proposal** describes a process surrounding Agorex or proposes a change to (or an event in) a process. Process Proposals often require community consensus and unlike Informational Proposals, they are more than recommendations, and users are typically not free to ignore them. Examples include procedures, guidelines, changes to the decision-making process, and changes to the tools or environment used in Agorex development.
+
+
```

Comment:
We should have a feature or exchange proposal outlined here as well.

Or are we keeping AIPs **only** for organizational matters?

Reply:
I would think this AIP process should be allowed for both, but with the exchange, speed of development is also important, so maybe a less formal process. (approved review of 1 board member for merges into master?)

Reply:
Ah, yes... This is what I was referring to above. As I wrote [here](https://github.com/Agorex-io/Proposals/pull/3/files#diff-cb86b9455b45e60922306503f26e5c70R102) I think we should consider optimizing the document for process/organizational proposals **only**. I think proposals for new "features" on the exchange should be done via pull requests and issues. For minor features, just do it (or at least start it) and submit a PR. For more significant changes, start an issue to discuss it and get community feedback first. 

I think the challenge for a community, open source exchange at first is actually getting enough actual contributors to develop the features, and ideas and agreement on features is much easier. Once we have so many contributors willing to develop features and the more evident problem is just deciding which ones to do, we could formalize the process by updating this doc (with a new AIP, or a PR updating this one).

Ethereum is a bit different in that Ethereum is a protocol and enhancements and changes _require_ of clients (similar to python). I think for the most part the exchange won't have that as such a significant concern.